### PR TITLE
feat(config): created objects for contract channel and governance config

### PR DESCRIPTION
**Pull request by jeff1010322**
Updated transactions to just have 2 types, call or config.

Permissioning is handled as type of config transactions and
contract updates are now included in contract channel config object.

### CODE REVIEW

File: idl/config.x [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+namespace mazzaroth
+{
+  // ContractChannelConfig stores contract channel configuration in state and is 
+  // accessible through host contract host functions
+  struct ContractChannelConfig
+  {
+    // Unique channel id which cannot be changed
+    ID channelID;
+    // Contract binary bytes
+    opaque contract<>;
```
Comment (kbartush) [inline idl/config.x]: When we discussed what should be in the contract config we used the rough criteria "Would we ever use this in a contract?"

I don't think we would ever use the contract bytes in a contract. Is this just an exception?

File: idl/config.x [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  enum ConsensusConfigType
+  {
+    NONE = 0,
+    PBFT = 1
+  };
+
+  // Permissioning types
+  enum PermissioningType
+  {
+    PUBLIC = 0,
```
Comment (kbartush) [inline idl/config.x]: Didn't we have a 3rd category of Permissioned? Maybe a bad name, but I think we need to account for the concept.

Comment (jeff1010322) [reply idl/config.x]: Yes, I have updated this enum.

File: idl/config.x (hunk 1 above)
Comment (jeff1010322) [reply idl/config.x]: After discussion, have replaced this with the hash and restored the contract update transaction type.

**Pull request merged**